swaybar/ipc.c

```diff
@@ -364,6 +380,30 @@ bool handle_ipc_event(struct bar *bar) {
 		json_object_put(result);
 		break;
 	}
+	case IPC_EVENT_MODIFIER: {
+		json_object *result = json_tokener_parse(resp->payload);
+		if (!result) {
+			free_ipc_response(resp);
+			sway_log(L_ERROR, "failed to parse payload as json");
+			return false;
+		}
+		json_object *json_change;
+		if (json_object_object_get_ex(result, "change", &json_change)) {
+			const char *change = json_object_get_string(json_change);
+			free(bar->config->hidden_state);
+			if (strcmp(change, "pressed") == 0) {
+				bar->config->hidden_state = strdup("show");
+			} else { //must be "released"
```

Comment:
defensive: assert instead of comment. Check for expected value and log unexpected values with `L_ERROR` instead?